### DISCUSSION TITLE
Spawn command in new tab question

### DISCUSSION THREAD

**Discussion by idr4n** [wezterm/wezterm#5889] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240701-070926-69686f45

### Ask your question!

I want to spawn a new tab and cd into a chosen directory using zoxide/fzf.

What I have tried so far, although it runs fzf, once I select the directory, the tab is closed right away:

```lua
{
  key = "/",
  mods = "CMD",
  action = wezterm.action_callback(function(window, _)
    window:mux_window():spawn_tab({
      -- args = { os.getenv("SHELL"), "-c", "cd $(zoxide query -l | fzf)" }, -- THIS DOESN'T WORK
      -- args = { "cd", os.getenv("HOME") .. "dotfiles" }, -- THIS DOESN'T WORK
      args = { os.getenv("SHELL"), "-c", os.getenv("HOME") .. "/bin/nvim" }, --THIS WORKS
    })
  end),
}
```

Now, with `args = { os.getenv("SHELL"), "-c", os.getenv("HOME") .. "/bin/nvim" },`, this works but once I close `nvim` then the tab is closed as well.

What I want is to run `zoxide query -l | fzf` to select the directory and cd into it in the new tab.

Thanks in advance!

**Comment by wez** [comment, marked as answer]
Look at the bottom example on https://wezfurlong.org/wezterm/config/lua/gui-events/gui-startup.html where it uses `pane:send_text` to "type" in the command.

**Reply by idr4n** [reply]
Yes!, `send_text` worked like a charm. Thanks a lot!

**Reply by skoch13** [reply]
@idr4n small suggestion from my side: you might not need to execute `zoxide query -l | fzf` as `zoxide` provides `zi` which does pretty much the same thing.
